server/src/main/java/org/elasticsearch/gateway/ClusterStateUpdaters.java

```diff
@@ -34,7 +34,10 @@ public static ClusterState setLocalNode(final ClusterState clusterState, Discove
             .build();
     }
 
-    static ClusterState upgradeAndArchiveUnknownOrInvalidSettings(final ClusterState clusterState, final ClusterSettings clusterSettings) {
+    public static ClusterState upgradeAndArchiveUnknownOrInvalidSettings(
```

Comment:
This is a bit unrelated, but we need access to this method in order to read the cluster state from the blob store.